apis/python/src/tiledbsoma/annotation_dataframe.py

```diff
@@ -76,10 +75,19 @@ def ids(self) -> Sequence[str]:
         """
         with self._open("r") as A:
             self.dim_name = A.domain.dim(0).name
-            # TileDB string dims are ASCII not UTF-8. Decode them so they readback
-            # not like `b"AKR1C3"` but rather like `"AKR1C3"`.
+
+            # TileDB string dims are ASCII not UTF-8. Decode them so they readback not like
+            # `b"AKR1C3"` but rather like `"AKR1C3"`. Update as of
+            # https://github.com/TileDB-Inc/TileDB-Py/pull/1304 these dims will read back OK.
             retval = A.query(attrs=[], dims=[self.dim_name])[:][self.dim_name].tolist()
-            return [e.decode() for e in retval]
+
+            retval = [e.decode() for e in retval]
+
+            if len(retval) > 0 and isinstance(retval[0], bytes):
+                return [e.decode() for e in retval]
+            else:
+                # list(...) is there to appease the linter which thinks we're returning `Any`
+                return list(retval)
```

Comment:
Could something like this work?

```
retval = A.query(attrs=[], dims=[self.dim_name])[:][self.dim_name]
return np.frombuffer(retval, dtype="U").tolist()
```

Reply:
no sorry @nguyenv !

```
    return np.frombuffer(retval, dtype="U").tolist()
ValueError: itemsize cannot be zero in type
```

Reply:
But regardless all I'm doing is putting an `if` around known-good code, invoking it when needed for older arrays which predate this PR!

Reply:
@nguyenv what works remains to accept or reject this PR?